src/content/post/20161122_drupal_application_bottlenecks_avoid_with_profiling.md

```diff
@@ -0,0 +1,152 @@
++++
+date = "2016-11-22T11:02:30Z"
+draft = false
+title = "Avoid bottlenecks on a Drupal web application and how identify it with profiling"
+description = "How identify bottleneck of your web application and avoid that can appear in the future"
+tags        = [ "profiling", "drupal", "performance", "web application"  ]
+topics      = [ "Profiling", "Drupal", "Performance" ]
+author = "Vinz"
++++
+
+### TL;DR
+
+This article is focused on the possible *bottlenecks of a web application*, how you can identify it to improve the performance of your code.
+
+We talk about profiling with *Blackfire*, PHP web app focused on *Drupal*. LAMP is the reference stack.
+
+<!-- more -->
+
+Application's performance is one of characteristics what separates a job from a good job, and demonstrates quality work.
+
+A customer's call about website performances, if server infrastructure is adequate to requests, most probably indicates some errors on our code.
+
+Google can be our friend if we find some 'web application performance checklist' but we have a problem, now. We need to identify and resolve our performance issues.
+
+A little disclaimer: this article is focused on a Drupal application, but information which follow are useful (I hope!) regardless of framework that you have used.
+
+### Application's check
```

Comment:
Sorry but here I hit a kind of stop. What am I reading from this moment on? A recipe? The googled checklist you mentioned before? Is this related to blackfire?

I think you should set a stage with a brief introduction like:

```
We'll follow a logical path from the first general check-up to the actual profiling, based on a real example.
```

Then you can move on this way:

* Present the example case (so everybody can follow along)
* Title each following paragraph so that it's clear what I am reading... What "Application's check" means? "Application-level checklist" is a bit more clear. 
* If you know you want to go through a sequence of more specific checks (from general to detail), write a brief in the introduction, like:

```
We'll follow a logical path from the first general check-up to the actual profiling, based on a real example.
We will:
* Set the stage
* List all possible checks we want to perform at application level
* List all possible checks we want to perform at server level
* etc etc...
```

This would help the reader value the post's content.